src/installer/installer.go

```diff
@@ -672,57 +673,71 @@ func (i *installer) cleanupInstallDevice() error {
 		devices, err = i.ops.GetRaidDevices(i.Device)
 
 		if err != nil {
-			return err
+			ret = utils.CombineErrors(ret, err)
 		}
 
 		for _, device := range devices {
 			// Cleaning the raid device itself before removing membership.
 			err = i.cleanupDevice(device)
 
 			if err != nil {
-				return err
+				ret = utils.CombineErrors(ret, err)
 			}
 		}
 
 		err = i.ops.CleanRaidMembership(i.Device)
 
 		if err != nil {
-			return err
+			ret = utils.CombineErrors(ret, err)
 		}
 		i.log.Infof("Finished cleaning up device %s", i.Device)
 	}
 
-	return i.ops.Wipefs(i.Device)
+	err = i.ops.Wipefs(i.Device)
+	if err != nil {
+		ret = utils.CombineErrors(ret, err)
+	}
+
+	return ret
 }
 
 func (i *installer) removeVolumeGroupsFromDevice(vgNames []string, device string) error {
+	var ret error
 	for _, vgName := range vgNames {
 		i.log.Infof("A volume group (%s) was detected on the installation device (%s) - cleaning", vgName, device)
 		err := i.ops.RemoveVG(vgName)
 
 		if err != nil {
 			i.log.Errorf("Could not delete volume group (%s) due to error: %w", vgName, err)
-			return err
+			ret = utils.CombineErrors(ret, err)
 		}
 	}
 
-	return nil
+	return ret
 }
 
 func (i *installer) cleanupDevice(device string) error {
+	var ret error
 	vgNames, err := i.ops.GetVolumeGroupsByDisk(device)
 	if err != nil {
-		return err
+		ret = err
 	}
 
 	if len(vgNames) > 0 {
-		err = i.removeVolumeGroupsFromDevice(vgNames, device)
-		if err != nil {
-			return err
+		if err = i.removeVolumeGroupsFromDevice(vgNames, device); err != nil {
+			ret = utils.CombineErrors(ret, err)
 		}
 	}
 
-	return i.ops.RemoveAllPVsOnDevice(device)
+	if err = i.ops.RemoveAllPVsOnDevice(device); err != nil {
```

Comment:
If this is successful we can ignore the previous error while removing the VG, no?

Reply:
We could, but I wanted to give all the information possible in case of a failure

Anyways, it would be pretty weird that pvremove works if the VG is not removed before ... I guess in that case I'd like to know anyways, because that's something funky